libs/services/Converter.py

```diff
@@ -20,31 +18,53 @@ def query_the_service(self, service, args, method='GET', data=None):
         :return: obtained response
         """
         try:
-            identification = f'{service}:{args}'
-            cached_result = self.cache.get(identification, None)
-            if cached_result:
-                return cached_result
-            result = self.execute_request(self.services[service] + args, method, data)
-            self.cache[identification] = result
+            result = await self.loop_request(self.services[service] + args, method, data)
             return result
-        except requests.exceptions.ConnectionError:
-            raise ConnectionError(f'Service {service} is not available')
+        except TypeError:
+            pass  # TODO: log - probably given argument is incorrect
 
-    def execute_request(self, url, method, data=None):
+    async def loop_request(self, url, method, data, depth=10):
         """
         Execute request with type depending on specified method.
 
         :param url: service URL
         :param method: GET/POST
         :param data: given arguments for POST request
+        :param depth: allowed recursion depth for unsuccessful requests
         :return: obtained response
         """
-        if method == 'GET':
-            return requests.get(url)
+        try:
+            if method == 'GET':
+                async with self.session.get(url=url) as response:
+                    return await self.process_request(response, url, method, data, depth)
+            else:
+                async with self.session.post(url=url, data=data) as response:
+                    return await self.process_request(response, url, method, data, depth)
+        except ServerDisconnectedError:
+            if depth > 0:
+                return await self.loop_request(url, method, data, depth - 1)
+
+    async def process_request(self, response, url, method, data, depth):
+        """
+        Method to wrap response handling (same for POST and GET requests).
+
+        :param response: given async response
+        :param url: service URL
+        :param method: GET/POST
+        :param data: given arguments for POST request
+        :param depth: allowed recursion depth for unsuccessful requests
+        :return: processed response
+        """
+        result = await response.text()
+        if response.ok:
+            return result
+        elif response.status == 503:
```

Comment:
Do you expect that repeating the same request 10 times is going to yield different results often enough to validate the extra load?

Reply:
For example using [PubChem](https://pubchemdocs.ncbi.nlm.nih.gov/dynamic-request-throttling), the service occasionally responses with code 503 when operating on close-to-limit requests frequencies. This will be improved and targeted for individual services in  #16 including requests/time limits.